src-brittany/Main.hs

```diff
@@ -128,29 +130,36 @@ mainCmdParser helpDesc = do
     when printHelp $ do
       liftIO $ print $ ppHelpShallow desc
       System.Exit.exitSuccess
-    inputPathM <- case maybeToList inputParam ++ inputPaths of
-      [] -> do
-        return Nothing
-      [x] -> return $ Just x
-      _   -> do
-        putStrErrLn $ "more than one input, aborting"
-        System.Exit.exitWith (System.Exit.ExitFailure 51)
-    outputPathM <- case outputPaths of
-      [] -> do
-        return Nothing
-      [x] -> return $ Just x
-      _   -> do
-        putStrErrLn $ "more than one output, aborting"
-        System.Exit.exitWith (System.Exit.ExitFailure 52)
+    when (length outputPaths > 0 && inplace) $ do
+      putStrErrLn "cannot specify output files and inplace at the same time"
+      System.Exit.exitWith (System.Exit.ExitFailure 52)
+
+    let inputPaths' = nonEmptyList Nothing . map Just $ maybeToList inputParam ++ inputPaths
+    let outputPaths' = if inplace then inputPaths' else nonEmptyList Nothing . map Just $ outputPaths
+    when (length inputPaths' /= length outputPaths') $ do
+      putStrErrLn "the number of inputs must match ther number of outputs"
+      System.Exit.exitWith (System.Exit.ExitFailure 51)
+
     config <- runMaybeT (readConfigs cmdlineConfig configPaths) >>= \case
       Nothing -> System.Exit.exitWith (System.Exit.ExitFailure 53)
       Just x  -> return x
     when (confUnpack $ _dconf_dump_config $ _conf_debug $ config) $ do
       trace (showConfigYaml config) $ return ()
-    eitherErrSucc <- coreIO putStrErrLn config suppressOutput inputPathM outputPathM
-    case eitherErrSucc of
-      Left errNo -> System.Exit.exitWith (System.Exit.ExitFailure errNo)
-      Right ()   -> pure ()
+
+    let ios = zipWith (coreIO putStrErrLn config suppressOutput) inputPaths' outputPaths'
+    errNoM <- foldM run Nothing ios
+    case errNoM of
+      Just errNo -> System.Exit.exitWith (System.Exit.ExitFailure errNo)
+      Nothing    -> pure ()
+ where
+  run acc io = do
+    res <- io
+    case res of
+      Left _   -> return (Just 1)
+      Right () -> return acc
+
+  nonEmptyList def [] = [def]
+  nonEmptyList _   x  = x
```

Comment:
In my opinion the case-of that was above before is nicer. This helper obscures the logic by forcing me to jump half a screen down and back up, with little benefit.

There is another source of redundancy I'd rather focus on: The code above effectively checks `inplace` twice and `null outputPaths` twice as well. Merging those two makes more sense imo.